pytorch/training/docker/1.9/py3/cu111/Dockerfile.gpu

```diff
@@ -0,0 +1,325 @@
+ && curl -L -o ~/miniconda.sh https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh \
+ && chmod +x ~/miniconda.sh \
+ && ~/miniconda.sh -b -p ${CONDA_PREFIX} \
+ && rm ~/miniconda.sh \
+ && ${CONDA_PREFIX}/bin/conda install -c conda-forge \
+    python=$PYTHON_VERSION \
+ && ${CONDA_PREFIX}/bin/conda install -y \
+    # conda 4.10.0 requires ruamel_yaml to be installed. Currently pinned at latest.
+    ruamel_yaml==0.15.100 \
+ && ${CONDA_PREFIX}/bin/conda install -y -c anaconda \
+    numpy \
+    ipython \
+    mkl \
+    mkl-include \
+    cython \
+    typing \
+    future \
+    "pyopenssl>=17.5.0" \
+ && conda install -y -c dglteam dgl-cuda11.1 \
+ && ${CONDA_PREFIX}/bin/conda clean -ya
+
+# Conda installs links for libtinfo.so.6 and libtinfo.so.6.2 both
+# Which causes "/opt/conda/lib/libtinfo.so.6: no version information available" warning
+# Removing link for libtinfo.so.6. This change is needed only for ubuntu 20.04-conda, and can be reverted
+# once conda fixes the issue
+RUN rm -rf /opt/conda/lib/libtinfo.so.6
+
+RUN ${CONDA_PREFIX}/bin/conda config --set ssl_verify False \
+ && pip install --upgrade pip --trusted-host pypi.org --trusted-host files.pythonhosted.org \
+ && ln -s ${CONDA_PREFIX}/bin/pip /usr/local/bin/pip3
+
+RUN conda install -c pytorch magma-cuda111==2.5.2 \
+ && conda install -c conda-forge \
+    opencv \
+ && conda install -y scikit-learn \
+    pandas \
+    h5py \
+    requests \
+    libgcc \
+ && conda clean -ya
+
+# Install libboost from source. This package is needed for smdataparallel functionality [for networking asynchronous IO].
+RUN wget https://sourceforge.net/projects/boost/files/boost/1.73.0/boost_1_73_0.tar.gz/download -O boost_1_73_0.tar.gz \
+  && tar -xzf boost_1_73_0.tar.gz \
+  && cd boost_1_73_0 \
+  && ./bootstrap.sh \
+  && ./b2 threading=multi --prefix=${CONDA_PREFIX} -j 64 cxxflags=-fPIC cflags=-fPIC install || true \
+  && cd .. \
+  && rm -rf boost_1_73_0.tar.gz \
+  && rm -rf boost_1_73_0 \
+  && cd ${CONDA_PREFIX}/include/boost
+
+WORKDIR /opt/pytorch
+
+# Copy workaround script for incorrect hostname
+COPY changehostname.c /
+COPY start_with_right_hostname.sh /usr/local/bin/start_with_right_hostname.sh
+
+WORKDIR /root
+
+# Uninstall torch and torchvision before installing the custom versions from an S3 bucket
+RUN pip install --no-cache-dir -U \
+    smdebug==${SMDEBUG_VERSION} \
+    smclarify \
+    "sagemaker>=2,<3" \
+    sagemaker-experiments==0.* \
+    sagemaker-pytorch-training \
+    --no-cache-dir fastai==1.0.61 \
+    "pyyaml>=5.4,<5.5" \
+    "awscli<2" \
+    psutil \
+    Pillow \
+    scipy \
+    pybind11 \
+    click \
+    mpi4py==3.0.3 \
+    cmake==3.18.2.post1 \
+    torchnet \
+    "cryptography>3.2" \
+ && pip uninstall -y torch \
+ && pip install --no-cache-dir -U ${PT_TRAINING_URL} \
+ && pip uninstall -y torchvision \
+ && pip install --no-deps --no-cache-dir -U ${PT_TORCHVISION_URL}
+
+# install metis
+RUN rm /etc/apt/sources.list.d/*
```

Comment:
Why is this necessary, and why does it need its own `RUN` statement, rather than being merged with the `RUN` statement for the `metis` installation below?

Reply:
Build was failing without this. Was facing this issue: https://forums.developer.nvidia.com/t/failed-to-fetch-https-developer-download-nvidia-com-compute-machine-learning-repos-ubuntu1804-x86-64-packages-gz/156287